### ISSUE TITLE
Publish command is generating hundreds of recursive nested directories of \PublishOutput\bin\[Config]

### ISSUE THREAD

**Issue by derekpec** [Microsoft/msbuild#2185]
When I publish my ASP.NET Core web application, hundreds of nested directories of "\bin\Release" are being generated in the output directory.  Eventually, after publishing my project multiple times, I started getting publishing failures with Visual Studio abruptly crashing ("Visual Studio has stopped working" with no error description whatsoever). I eventually determined the cause of the crash was because the filepaths had become too long, caused by this issue.  This issue does not happen when the `build` command is run by itself, only when running `publish`.  My project uses ASP.NET Core 1.0.1 .NET Framework 4.5 (`portable-net45+win8`); I'm using Visual Studio 2015; and I'm trying to publish to my local filesystem.

Here is a snippet from my build output following a successful publish, but still showing this issue (as opposed to the many crashes, which happened 100% of the time until I changed the output directory to artificially and temporarily shorten the filepaths):
```
------ Publish started: Project: SmartConfirmResponses, Configuration: Release Any CPU ------
Connecting to C:\Temp\Publish...
rmdir /S /Q "C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\"
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\Views - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Debug\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\Views\Home\Index.cshtml - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\Views\Shared\DataFilter.cshtml - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\Views\Shared\Error.cshtml - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\Views\Shared\_Layout.cshtml - The system cannot find the path specified.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~1\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\bin\Release\PUBLIS~3\bin\Release\PUBLIS~1\bin\Release\PUBLIS~2\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1\bin\Release\PUBLIS~1 - The directory is not empty.
Environment variables:
Path=.\node_modules\.bin;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Web\External;%PATH%;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Web\External\git
C:\Program Files\dotnet\dotnet.exe publish "C:\Users\derek\Documents\Visual Studio 2015\Projects\PEC\SmartConfirm\SmartConfirmResponses" --framework netcoreapp1.0 --output "C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122" --configuration Release --no-build
Publishing SmartConfirmResponses for .NETCoreApp,Version=v1.0
Bundling with configuration from C:\Users\derek\Documents\Visual Studio 2015\Projects\PEC\SmartConfirm\SmartConfirmResponses\bundleconfig.json
Processing wwwroot/css/site.min.css
  Bundled
  Minified
Processing wwwroot/js/site.min.js
  Bundled
  Minified
Configuring the following project for use with IIS: 'C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122'
Updating web.config at 'C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122\web.config'
Configuring project completed successfully
publish: Published to C:\Users\derek\AppData\Local\Temp\PublishTemp\SmartConfirmResponses122
Published 1/1 projects successfully
Publishing with publish method [FileSystem]
Publishing files to C:\Temp\Publish
Executing command ["C:\Program Files (x86)\IIS\Microsoft Web Deploy V3\msdeploy.exe" -source:manifest='C:\Users\derek\AppData\Local\Temp\PublishTemp\obj\SmartConfirmResponses122\SourceManifest.xml' -dest:manifest='C:\Users\derek\AppData\Local\Temp\PublishTemp\obj\SmartConfirmResponses122\DestinationManifest.xml' -verb:sync -retryAttempts:20 -disablerule:BackupRule]
Info: Adding file (C:\Temp\Publish\appsettings.json).
Info: Adding file (C:\Temp\Publish\appsettings.Production.json).
Info: Adding directory (C:\Temp\Publish\bin).
Info: Adding directory (C:\Temp\Publish\bin\Debug).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin).
Info: Adding directory (C:\Temp\Publish\bin\Debug\PublishOutput\bin\Debug\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release\PublishOutput\bin\Release).
...
```

**Comment by derekpec** [comment]
Due to the publishing failure and VS crashing, I initially tried to change my output directory target location from `.\bin\Release\PublishOutput` to `C:\temp\Publish`, in an effort to shorten the filepaths, per the recommendations in https://github.com/Microsoft/dotnet/issues/234 and https://github.com/aspnet/Home/issues/790#issuecomment-306723506.  Sometimes this work-around works for me, sometimes it doesn't.  The underlying issue is clearly the incorrect generation of all these nested directories.

**Comment by derekpec** [comment]
In addition to the above partially-working work-around, I have tried:
- wiping my Publishing Profiles and creating new ones, multiple times.
- wiping my project.json file and re-creating it
- updating the "publishOptions" of my project.json file to:
`"publishOptions": {
    "exclude": [
      "bin",
      "node_modules"
    ],
    "include": [
      "wwwroot",
      "**/*.cshtml",
      "appsettings.json",
      "web.config"
    ],
    "publishExclude": [ "bin", "**.user", "**.vspscc" ]
  },`

Then, I went back to look at earlier versions of my project that I had published and saved copies of and found that this issue has actually been present from my first publish!  It appears the nested directories get deeper and longer each additional publish, perhaps?

I wonder how many others have this issue and are running into max filepath warnings as described in the two issues referenced in my previous comment and https://github.com/Microsoft/msbuild/issues/53.

**Comment by derekpec** [comment]
I tried the "fixes" described at https://github.com/aspnet/Tooling/issues/927#issuecomment-270519625 and https://github.com/dotnet/sdk/issues/207#issuecomment-252261547, with no success. 

Here is my project.json:

```
{
  "dependencies": {
    "Microsoft.NETCore.App": {
      "version": "1.0.1",
      "type": "platform"
    },
    "Microsoft.AspNetCore.Diagnostics": "1.0.0",
    "Microsoft.AspNetCore.Mvc": "1.0.1",
    "Microsoft.AspNetCore.Razor.Tools": {
      "version": "1.0.0-preview2-final",
      "type": "build"
    },
    "Microsoft.AspNetCore.Routing": "1.0.1",
    "Microsoft.AspNetCore.Server.IISIntegration": "1.0.0",
    "Microsoft.AspNetCore.Server.Kestrel": "1.0.1",
    "Microsoft.AspNetCore.StaticFiles": "1.0.0",
    "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.0.0",
    "Microsoft.Extensions.Configuration.Json": "1.0.0",
    "Microsoft.Extensions.Logging": "1.1.1",
    "Microsoft.Extensions.Logging.Console": "1.0.0",
    "Microsoft.Extensions.Logging.Debug": "1.0.0",
    "Microsoft.Extensions.Options.ConfigurationExtensions": "1.0.0",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0",
    "Microsoft.EntityFrameworkCore.SqlServer": "1.1.2"
  },
  "tools": {
    "BundlerMinifier.Core": "2.0.238",
    "Microsoft.AspNetCore.Razor.Tools": "1.0.0-preview2-final",
    "Microsoft.AspNetCore.Server.IISIntegration.Tools": "1.0.0-preview2-final",
    "Microsoft.EntityFrameworkCore.Tools": "1.0.0-preview2-final"
  },
  "frameworks": {
    "netcoreapp1.0": {
      "imports": [
        "dotnet5.6",
        "portable-net45+win8"
      ]
    }
  },
  "buildOptions": {
    "emitEntryPoint": true,
    "preserveCompilationContext": true,
    "xmlDoc": false
  },
  "runtimeOptions": {
    "configProperties": {
      "System.GC.Server": true
    }
  },
  "publishOptions": {
    "exclude": [
      "bin/**",
      "obj/**",
      "node_modules",
      "**.user",
      "**.vspscc"
    ],
    "include": [
      "wwwroot",
      "Views",
      "appsettings.json",
      "web.config"
    ]
  },
  "scripts": {
    "prepublish": [ "bower install", "dotnet bundle" ],
    "postpublish": [ "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%" ]
  }
}
```

**Comment by rainersigwald** [comment]
@derekpec Can you clarify whether you're using MSBuild or the older preview tooling? You're posting only a `project.json`, which is not consumed by MSBuild.

**Comment by derekpec** [comment]
@rainersigwald - Sorry, I guess I was using the older tooling with `project.json`/`.xproj`, and not `MSBuild`/`.csproj`. I had [incorrectly] assume `build` was a wrapper for `msbuild` and didn't realize the two were different.  I'm still learning ASP.NET Core and trying to keep up with all the recent changes/updates. I thought to start with VS 2015 to avoid all the [potential] new bugs in VS 2017, but now I see that the default templating (e.g. with `project.json`) is already outdated.  @eerhardt helped point me in the right direction to migrate, which fixed (or, at least, bypassed) the issue: https://github.com/dotnet/sdk/issues/207#issuecomment-307406173.  Thanks.  I guess since `project.json` is no longer supported (that's _sooo_ 2016!), I'll close this issue.  Thanks again to both of you.

**Comment by rainersigwald** [comment]
> I had [incorrectly] assume `build` was a wrapper for `msbuild` and didn't realize the two were different.

In your defense, that's true _now_, just not with the older tooling. You were just ahead of your time 😁

**Comment by derekpec** [comment]
If anyone who reads this and may still be dealing with this issue, here's a pro tip:  skip the `bin` directory when you copy your files to your deployment server.  Even when I have been able to successfully publish my project without a VS 2015 crash (using the workaround of setting the publish directory in the publish profile to something short like `C:\temp`), the filepaths in `bin` are too long to copy.  As is all too common, there was no error message generated when I tried to transfer them, it simply "didn't work". Lo and behold, leave out the `bin` folder and it "just works" as expected.  Clearly this is an issue with the destination filepath being too long, although I wish we could get a [descriptive] error message indicating such.
